### DISCUSSION TITLE
Add page with contact information

### DISCUSSION THREAD

**Discussion by msvoelker** [buildbot/buildbot#7617] [Q&A]
Hi,

I'm using buildbot version 3.11.1. Is there a simple way to add a page with static contact information, or just a link (e.g., in the sidebar-footer) to an external page?

My buildbot runs on a FreeBSD machine. I installed it from the ports collection with pkg install py39-buildbot py39-buildbot-www. By simple, I mean preferable without building buildbot from sources, which means without writing a full plugin I guess.

I tried to set custom_templates_dir but couldn't figure out what the directory should contain to change anything.

Thanks,

Timo

**Comment by vibbo** [comment]
Hi.

You can have a look at UI customization done via [wsgi_dashboards](https://docs.buildbot.net/latest/manual/customization.html#writing-dashboards-with-flask-or-bottle) plugin .

As an example you can inspect [MariaDB buildbot sponsor dashboard](https://buildbot.mariadb.org/#/sponsor), and it's [sources](https://github.com/MariaDB/buildbot/blob/f966a2ba4c7b79b63c7f015a691b1473a6aa8b67/master-web/master.cfg#L73C22-L73C37).

Hope this helps.

**Comment by msvoelker** [comment]
Thanks for your answer. I'm running the buildbot master on FreeBSD and installed it with

    pkg install py311-buildbot py311-buildbot-www ...

A package for the wsgi_dashboards plugin does not exist. I installed that with

    pip install buildbot_wsgi_dashboards

For a first test, I just added the plugin to the list of plugins in my master.cfg
    
    c['www'] = dict(port=config['webgui_proto']+':'+str(config['webgui_port']),
                logfileName='/dev/null',
                plugins=dict(waterfall_view={}, console_view={}, grid_view={}, wsgi_dashboards={}))

When I now open the buildbot master webpage, I see nothing but the following errors in Firefox's Web Developer Tools.

```
GET
http://buildbot.nplab.de:48010/wsgi_dashboards/styles.css?_1719257192259
[HTTP/1.1 404 Not Found 89ms]

Uncaught TypeError: f is undefined
    <anonymous> http://buildbot.nplab.de:48010/wsgi_dashboards/scripts.js?_1719257192259:9
    <anonymous> http://buildbot.nplab.de:48010/wsgi_dashboards/scripts.js?_1719257192259:1
    <anonymous> http://buildbot.nplab.de:48010/wsgi_dashboards/scripts.js?_1719257192259:1
[scripts.js:9:109](http://buildbot.nplab.de:48010/wsgi_dashboards/scripts.js?_1719257192259)
Uncaught Error: [$injector:modulerr] Failed to instantiate module app due to:
[$injector:modulerr] Failed to instantiate module wsgi_dashboards due to:
[$injector:nomod] Module 'wsgi_dashboards' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.


[https://errors.angularjs.org/1.8.0/$injector/modulerr?...
    Angular 9
    jQuery 13
    Webpack 12
[angular.js:138](webpack://buildbot-www/guanlecoja-ui/node_modules/angular/angular.js)
```

What am I doing wrong?

PS: During my tests, I updated to buildbot 3.11.5.
